### ISSUE TITLE
request access: when a user requests access to a file and is rejected, that action is saved and should be available via API

### ISSUE THREAD

**Issue by sbarbosadataverse** [IQSS/dataverse#8013]
At this time, rejected users are not saved in any way on the file level permissions page:
1. This removes and doesn't allow the dataset owner to track their actual numbers of data requests on their restricted file/dataset
2. This doesn't provide the user with a list of previous requests they can track and keep a record of rejected users
3. This has been requested in support tickets into Harvard Dataverse repeatedly.

**Comment by sbarbosadataverse** [comment]
@TaniaSchlatter I created this ticket for the request access-rejected user scenario

**Comment by qqmyers** [comment]
FWIW: The database is tracking granted and rejected requests these days, but there's no UI/API way to retrieve them yet. It would be relatively easy to add a parameter to the https://guides.dataverse.org/en/latest/api/dataaccess.html#list-file-access-requests call, e.g. ?includeHistory=true (mirroring the flag name from #11268 for curationStatus) to at least make the info accessible. It should also be possible to add the info to the current UI or SPA, but some guidance would be needed as to where to display it (showing the history for all files in the dataset page (permissions tab) could be a long list, could show it only on the file page? or?).

**Comment by pdurbin** [comment]
Right, as discussed [in Slack](https://iqss.slack.com/archives/C03R1E7T4KA/p1740421436528709?thread_ts=1740417754.612809&cid=C03R1E7T4KA) as of the following PR there are three possible RequestState values that can be saved in the database: CREATED, GRANTED, REJECTED.

- #9599

Yes, it would be great to expose this via API.

**Comment by cmbz** [comment]
2025/03/03
- Would require a backend db change
- Would require UI development

**Comment by cmbz** [comment]
2025/03/03
- Moved to dv backlog column @scolapasta

**Comment by scolapasta** [comment]
Ah, I hadn't realized we were now storing rejected. So yes, if that's the case, no DB change; still needs UI design (especially to handle when large number of rejects exist).

**Comment by cmbz** [comment]
2025-04-09
Next steps for whomever picks up issue
- Confirm APIs don't exist
- Implement support in the API
- Open an SPA issue for the UI part (design and implementation)
Sizing at 30 initially

**Comment by pdurbin** [comment]
Based on @cmbz's comment above I retitled this issue.

Also, @DieuwertjeBloemen just opened the following related issue, after a discussion at https://dataverse.zulipchat.com/#narrow/channel/378866-troubleshooting/topic/file.20access.20requests.20in.20a.20collection

- #11746

**Comment by stevenwinship** [comment]
Looking for UI comments: ( I still need to consider pagination )

Original without showing history

<img width="1137" height="177" alt="Image" src="https://github.com/user-attachments/assets/c91e05b8-3a12-471c-83f9-2447c0a52149" />

New showing history

<img width="1146" height="258" alt="Image" src="https://github.com/user-attachments/assets/be35aed1-8ad0-4f51-b2a2-be35fd6e8128" />

<img width="1143" height="244" alt="Image" src="https://github.com/user-attachments/assets/2b0d44c8-11c4-4d27-bb83-d90437f1b1b6" />

**Comment by pdurbin** [comment]
> * Open an SPA issue for the UI part (design and implementation)

I'm confused. I thought the plan was to work on the UI in the SPA rather than JSF, per above.

**Comment by stevenwinship** [comment]
I was going off Jim's comment "It should also be possible to add the info to the current UI or SPA".
If that's not relevant then I can remove it

**Comment by stevenwinship** [comment]
@pdurbin I'm not sure what the frown face emoji indicates. Should I remove the JSF change or leave it in?

**Comment by pdurbin** [comment]
@stevenwinship ha! In Firefox, at least, text pops up that says "confused", not "frown". 😄

Thanks for discussing at standup today. I'm not sure if we all reached a consensus or not. 🤷

**Comment by stevenwinship** [comment]
Created SPA Issue https://github.com/IQSS/dataverse-frontend/issues/895

**Comment by stevenwinship** [comment]
Now that we can see the history of requests I noticed something that may or may not make sense.

A user requests access to a file and that access is granted. The history shows the state of the request as being granted.
When the access is then removed the request still shows a state of granted. Should the request be updated to show rejected? 
It makes sense that the access being removed doesn't have anything to do with the request but it seems odd when viewing both in the UI and having them contradict each other.

**Comment by pdurbin** [comment]
Hmm, should it show both? Accepted as of a timestamp and rejected as of another timestamp? 🤔

**Comment by stevenwinship** [comment]
Removing access is not in the fileaccessrequests table since it's not a request. 
It's removing the file_downloader role. the method (removeRoleAssignments()) doesn't check for existing requests like the grantAccessToRequests() does

**Comment by qqmyers** [comment]
I don't think revoking access later represents rejecting the request. There is a separate history for roleaccess that should show when access was assigned and removed.
